File: smart_freight.client/src/types/truck.ts
export type Truck = {
    id: string;
    name: string;
    plateNumber: string;
    maxPayloadKg: number;
    isActive: boolean;
    createdAt: string;
};

export type TruckCreateRequest = {
    name: string;
    plateNumber: string;
    maxPayloadKg: number;
    isActive: boolean;
};

export type TruckUpdateRequest = TruckCreateRequest;

File: Smart_Freight.Server/Models/Truck.cs
namespace Smart_Freight.Server.Models;

public class Truck
{
    public Guid Id { get; set; }
    public string Name { get; set; } = string.Empty;
    public string PlateNumber { get; set; } = string.Empty;
    public decimal MaxPayloadKg { get; set; }
    public bool IsActive { get; set; } = true;
    public DateTimeOffset CreatedAt { get; set; } = DateTimeOffset.UtcNow;
}
